File: app/channels/cepar_feed_channel.rb
class CeparFeedChannel < ApplicationCable::Channel
  def subscribed
    stream_from "cepar_feed_channel"
  end

  def talk(data)
    message = data["content"]
    ActionCable.server.broadcast("cepar_feed_channel", { content: message })
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/cepar_feed_channel.js
import consumer from "channels/consumer"

const ceparFeedChannel = consumer.subscriptions.create("CeparFeedChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Connected to CeparFeedChannel.");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log("Disconnected from CeparFeedChannel.");
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    const messagesDiv = document.getElementById("messages");
    if (messagesDiv && data.content) {
      const messageElement = document.createElement("p");
      messageElement.textContent = data.content;
      messagesDiv.appendChild(messageElement);
    }
  }
});


function sendMessage(content) {
  ceparFeedChannel.perform("talk", { content: content });
}

export { sendMessage };
window.sendMessage = sendMessage;